vhdl/quartus/quartus_workflow.py

```diff
@@ -0,0 +1,346 @@
+                    return True
+            return False
+    except FileNotFoundError:
+        return False
+
+
+def gen_project(project_name, target, template, working_dir):
+    """Generates and compiles the project defined in make_project.tcl
+
+    Parameters
+    ----------
+    working_dir : string
+        Working directory of the generation process
+    """
+    gen_project_tcl(project_name, target, template, working_dir)
+    if(target.name == "arria10"):
+        gen_pll_qsys(working_dir)
+
+    log_msg = "Generating project"
+    cmd = f"cd {working_dir} && {QUARTUS_BIN_DIR}quartus_sh -t make_project.tcl"
+    log_file_path = working_dir + "project_gen.log"
+
+    run_cmd_and_log(cmd, log_msg, log_file_path)
+
+def compile_project(project_name, project_revision, template, working_dir):
+    tcl_file = "compile_project.tcl"
+    if not(os.path.isfile(working_dir + tcl_file)):
+        with open(working_dir + tcl_file, "w") as compile_file:
+            compile_file.write(template.add_quartus_compile_project(
+                project_name, project_revision))
+
+    log_msg = "Compiling project"
+    cmd = f"cd {working_dir} && {QUARTUS_BIN_DIR}quartus_sh -t {tcl_file}"
+    log_file_path = working_dir + "compile_project.log"
+
+    run_cmd_and_log(cmd, log_msg, log_file_path)
+
+def gen_rbf(working_dir, target_system):
+    """Converts the sof file to an rbf file
+
+    Parameters
+    ----------
+    working_dir : string
+        Working directory of the generation process
+    target_system : string
+        The target system being generated
+    """
+    log_msg = "Generating rbf file"
+    # Command uses -m FPP for the fast passive parallel which ends up being equivalent to passive parallel x16
+    cmd = f"cd {working_dir}/output_files && {QUARTUS_BIN_DIR}quartus_cpf -c -m FPP {target_system}.sof {target_system}.rbf"
+    log_file_path = working_dir + "rbf_gen.log"
+
+    run_cmd_and_log(cmd, log_msg, log_file_path)
+
+
+def execute_quartus_workflow(config, working_dir=""):
+    """Executes quartus workflow that creates a system and project, compiles it,
+        and converts the output file to an RBF
+
+    Parameters
+    ----------
+    config : InputStructure
+        User input structure describing the intended generation  
+    working_dir : str, optional
+        Working directory of the generation process, by default ""
+    """
+    target = None
+
+    if(config.target_system == "de10"):
+        target = DE10
+    elif(config.target_system == "audioblade"):
+        target = Audioblade
+    else:
+        raise ValueError(
+            f"The provided target: {config.target_system} is not supported")
+
+    abspath = os.path.abspath(__file__)
+    dname = os.path.dirname(abspath)
+    global RES_DIR
+    RES_DIR = os.path.dirname(abspath) + "/res/"
+    global QUARTUS_BIN_DIR
+    global QSYS_BIN_DIR
+    QUARTUS_BIN_DIR = os.environ["QSYS_ROOTDIR"].split("sopc_builder")[
```

Comment:
Is `QUARTUS_ROOTDIR_OVERRIDE` defined on your Windows system? If so, we can use that environment variable instead of having to split part of the `QSYS_ROOTDIR`

Reply:
It isn't, although I do have `QUARTUS_ROOTDIR`. I avoided using it because my Quartus installation on Linux only made `QSYS_ROOTDIR` available. Using `QSYS_ROOTDIR` entirely enables it to be consistent across Linux and Windows.